src/Microsoft.Extensions.FileProviders.Physical/PollingWildCardChangeToken.cs

```diff
@@ -0,0 +1,153 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Security.Cryptography;
+using System.Text;
+using Microsoft.Extensions.FileProviders.Physical.Internal;
+using Microsoft.Extensions.FileSystemGlobbing;
+using Microsoft.Extensions.FileSystemGlobbing.Abstractions;
+using Microsoft.Extensions.Primitives;
+
+namespace Microsoft.Extensions.FileProviders.Physical
+{
+    public class PollingWildCardChangeToken : IChangeToken
+    {
+        public static readonly TimeSpan PollingInterval = TimeSpan.FromSeconds(4);
+        private const int DefaultBufferSize = 256;
+        private readonly object _enumerationLock = new object();
+        private readonly DirectoryInfoBase _directoryInfo;
+        private readonly Matcher _matcher;
+        private bool _changed;
+        private DateTime _lastScanTimeUtc;
+        private byte[] _fileNameBuffer;
+        private byte[] _previousHash;
+
+        public PollingWildCardChangeToken(
+            string root,
+            string pattern)
+            : this(
+                new DirectoryInfoWrapper(new DirectoryInfo(root)),
+                pattern,
+                Internal.Clock.Instance)
+        {
+        }
+
+        public PollingWildCardChangeToken(
+            DirectoryInfoBase directoryInfo,
+            string pattern,
+            IClock clock)
+        {
+            _directoryInfo = directoryInfo;
+            Clock = clock;
+
+            _matcher = new Matcher(StringComparison.OrdinalIgnoreCase);
+            _matcher.AddInclude(pattern);
+            CalculateChanges();
+        }
+
+        /// <inheritdoc />
+        public bool ActiveChangeCallbacks => false;
+
+        public IClock Clock { get; }
+
+        /// <inheritdoc />
+        public bool HasChanged
+        {
+            get
+            {
+                if (_changed)
+                {
+                    return _changed;
+                }
+
+                if (Clock.UtcNow - _lastScanTimeUtc >= PollingInterval)
+                {
+                    lock (_enumerationLock)
+                    {
+                        _changed = CalculateChanges();
+                    }
+                }
+
+                _lastScanTimeUtc = Clock.UtcNow;
+                return _changed;
+            }
+        }
+
+        private bool CalculateChanges()
+        {
+            var result = _matcher.Execute(_directoryInfo);
+            var hasChanges = false;
+
+            var files = result.Files.OrderBy(f => f.Path, StringComparer.OrdinalIgnoreCase);
```

Comment:
This can be case sensitive sort (default comparer). If case changed, the hash will be different anyways and order won't matter.